services/supervisor/src/transactions.rs

```diff
@@ -131,6 +128,47 @@ impl StopService {
     }
 }
 
+impl ResumeService {
+    fn validate(&self, context: &CallContext<'_>) -> Result<(), ExecutionError> {
+        let instance = get_instance(context, self.instance_id)?;
+
+        match instance.status {
+            Some(InstanceStatus::Stopped) => {}
+            _ => {
+                return Err(
+                    ConfigurationError::MalformedConfigPropose.with_description(format!(
+                        "Discarded an attempt to resume not stopped service instance: {}",
+                        instance.spec.name
+                    )),
+                )
+            }
+        }
+
+        if instance.data_version() != &self.artifact.version {
+            return Err(
+                ConfigurationError::MalformedConfigPropose.with_description(format!(
+                    "Discarded an attempt to resume service with incorrect artifact version: \
+                     got {}, expected {}",
+                    self.artifact.version,
+                    instance.data_version()
+                )),
+            );
+        }
+
+        if instance.spec.artifact.name != self.artifact.name {
```

Comment:
Nit: It could make sense to use [`ArtifactId::is_upgrade_of`](https://github.com/exonum/exonum/blob/616622f3b7e7bf7858590bae6b11732704fcfd7c/exonum/src/runtime/types.rs#L201) to be more forward-compatible. OTOH, the error message will be less specific in this case.

Reply:
`is_upgrade_of` has a little bit different behaviour.